app/slimming/slimming.js

```diff
@@ -156,6 +158,7 @@ app.controller('GOSlimCtrl', function($scope, $location, $q,
     delete $scope.selection[termToRemove.aspect].terms[termToRemove.id];
     // Add to de-selected items
     $scope.deSelectedItems.push(termToRemove);
+    $scope.testIfSelectionIsEmpty();
```

Comment:
These are the lines you would need to add at the end of the watch function:
```
if ($scope.count.total === 0){
  $scope.deSelectedItems = [];
}
```

But, I am not sure whether we really need a watch function at all. I am concern about performance. Every time we modify the selection we count all its items again. See extract from https://docs.angularjs.org/api/ng/type/$rootScope.Scope

>When objectEquality == true, inequality of the watchExpression is determined according to the angular.equals function. To save the value of the object for later comparison, the angular.copy function is used. This therefore means that watching complex objects will have adverse memory and performance implications.